boards/communication/src/data_manager.rs

```diff
@@ -52,74 +34,55 @@ impl DataManager {
         return RadioRate::Slow;
     }
 
-    /// Do not clone instead take to reduce CPU load.
-    pub fn take_sensors(&mut self) -> [Option<Message>; 13] {
-        [
-            self.air.take(),
-            self.ekf_nav_1.take(),
-            self.ekf_nav_2.take(),
-            self.ekf_nav_acc.take(),
-            self.ekf_quat.take(),
-            self.imu_1.take(),
-            self.imu_2.take(),
-            self.utc_time.take(),
-            self.gps_vel.take(),
-            self.gps_vel_acc.take(),
-            self.gps_pos_1.take(),
-            self.gps_pos_2.take(),
-            self.gps_pos_acc.take(),
-        ]
-    }
-
-    pub fn clone_states(&self) -> [Option<StateData>; 1] {
-        [self.state.clone()]
-    }
-    pub fn handle_data(&mut self, data: Message) {
-        match data.data {
-            messages::Data::Sensor(ref sensor) => match sensor.data {
-                messages::sensor::SensorData::EkfNavAcc(_) => {
-                    self.ekf_nav_acc = Some(data);
-                }
-                messages::sensor::SensorData::GpsPosAcc(_) => {
-                    self.gps_pos_acc = Some(data);
-                }
-                messages::sensor::SensorData::Air(_) => {
-                    self.air = Some(data);
-                }
-                messages::sensor::SensorData::EkfNav1(_) => {
-                    self.ekf_nav_1 = Some(data);
+    pub fn stuff_messages(&mut self) -> Result<Vec<u8, 255>, postcard::Error> {
+        let mut bytes: Vec<u8, 255> = Vec::new();
+        for el in self.message_queue.oldest_ordered() {
+            match el.data {
+                messages::Data::Command(_) => {
+                    if bytes.len() + MAX_COMMAND_SIZE <= MAX_SIZE {
```

Comment:
Can someone check this logic.

Reply:
Note: this is not optimal, but rather practical since we don't want to spend too much time deciding what fits and we just want it to get out of the system.